src/Parentheses.cpp

```diff
@@ -1,6 +1,31 @@
 #include "Parentheses.hpp"
+#include <stack>
 
 bool Parentheses::isBalanced(std::string s) {
-   
-   return (s!=")(");
-}
\ No newline at end of file
+
+int i = 0;
+std::stack<char> ParenthesesStack{};
+
+for(auto p = s.begin() ; i < s.size() ; p++, i++) {
+    if(ParenthesesStack.size() > 0) {
+        if (*p == '}') {
+            if(ParenthesesStack.top() == '{')
+                ParenthesesStack.pop();
+        }
+        else if (*p == ')') {
+            if(ParenthesesStack.top() == '(')
+                ParenthesesStack.pop();
+        }
+        else if (*p == ']') {
+            if(ParenthesesStack.top() == '[')
+                ParenthesesStack.pop();
+        }
```

Comment:
przy takim rozwiązaniu trzeba dopisywać nowego ifa dla każdego nowego elementu. Spróbujcie użyć kontenerów do przechowywania nawiasów, aby wyszukiwac ich w kontenerach. Dzięki temu jak trzeba będzie coś później dodać to nie trzeba będzie zmieniać logiki, tylko wrzuci się nowe rzeczy do kontenerów i zadziała z automatu.

Reply:
to moze taka propozycja bylaby lepsza? (nawiasy w jednym miejscu)

bool Parentheses::isBalanced(std::string s) {
    std::string brecketsClose =")]}";
    if (!s.empty() && s.size()>1)
    {
        while (s.size()>1 )
        {
            auto it = std::find_first_of(s.begin(), s.end(), brecketsClose.begin(), brecketsClose.end());
            if ( (s.size()>1 && static_cast<int>(*it) == static_cast<int>(*(it-1))+1) || (s.size()>1 &&(static_cast<int>(*it) == static_cast<int>(*(it-1))+2)) )
            {
                s.erase(it);
                s.erase(it-1);
            }
            else return false;    
        }
    }
    return s.empty();
}